### ISSUE TITLE
Inconsistent behaviour with hazelcast configuration

### ISSUE THREAD

**Issue by snicoll** [hazelcast/hazelcast#5414]
TL;DR sample project available [here](https://dl.dropboxusercontent.com/u/623579/hazelcast-jcache-config.zip)

Hazelcast is inconsistent whether it is started as a regular `HazelcastInstance` or as a compliant JSR-107 `CacheManager`.

The attached project configures these two means of retrieving a cache on the same `hazelcast.xml` file. While the `HazelcastCacheManager` will return the cache that is defined in configuration, the `javax.cache.CacheManager` counter-part will simply return `null` (?)

There are two tests in the sample project above. Thanks.

(Unrelated note: from the numerous discussion I had with you guys, I understood that Hazelcast will create a cache on the fly if it does not exist yet. Why is that not the case here? Note that it is fortunate because it helped me to spot that issue)

**Comment by serkan-ozal** [comment]
Hi @snicoll,

Thanks for the sample project that reproduce the case.

In your `hazelcast.xml`, you don't have a cache definition but have a map definition.
Hazelcast JCache API doesn't deal with map definitions for caches like Hazelcast's Spring cache manager. It uses cache configuration (`CacheConfig` as programmatic configuration and `<cache>` for declarative configuration)

So you must add cache configuration definition to your `hazelcast.xml`

``` xml
<cache name="countries">
    ....
</cache>
```

See http://docs.hazelcast.org/docs/latest/manual/html/jcache-jcacheconfiguration.html for more details.

I am closing this one, since it is not an issue.

Regards.

**Comment by snicoll** [comment]
Does the other way around works? If I specify a `cache` entry, will the Spring's cache manager use it? (i.e. native `HazelcastInstance`).

**Comment by serkan-ozal** [comment]
@snicoll No, Hazelcast Spring cache manager is based on `IMap` so it is not configured with `<cache>` configuration in xml
